devolutions-gateway/src/api/ai/mod.rs

```diff
@@ -0,0 +1,559 @@
+//! AI Router module (experimental)
+//!
+//! This module provides a proxy for AI provider APIs, currently supporting Mistral AI,
+//! Ollama, LM Studio, Anthropic, OpenAI, OpenRouter, and Azure OpenAI.
+//! It handles authentication via a gateway API key and forwards requests to the
+//! configured AI provider endpoints.
+
+mod provider;
+
+use std::time::Duration;
+
+use axum::Router;
+use axum::body::Body;
+use axum::extract::{Path, State};
+use axum::http::{HeaderMap, Request};
+use axum::response::Response;
+use axum::routing::{get, post};
+use http_body_util::BodyExt as _; // into_data_stream
+
+use crate::DgwState;
+use crate::http::HttpError;
+
+pub use provider::{AuthMethod, ProviderConfig, ProviderConfigBuilder};
+
+/// Anthropic API version header value
+const ANTHROPIC_API_VERSION: &str = "2023-06-01";
+
+/// Creates the AI router with all provider endpoints.
+pub fn make_router<S>(state: DgwState) -> Router<S> {
+    Router::new()
+        // Mistral AI routes
+        .route("/mistral/v1/models", get(mistral_list_models))
+        .route("/mistral/v1/models/{model_id}", get(mistral_get_model))
+        .route("/mistral/v1/chat/completions", post(mistral_chat_completions))
+        // OpenAI routes
+        .route("/openai/v1/models", get(openai_list_models))
+        .route("/openai/v1/chat/completions", post(openai_chat_completions))
+        // Ollama routes
+        .route("/ollama/v1/models", get(ollama_list_models))
+        .route("/ollama/v1/chat/completions", post(ollama_chat_completions))
+        // LM Studio routes
+        .route("/lmstudio/v1/models", get(lmstudio_list_models))
+        .route("/lmstudio/v1/chat/completions", post(lmstudio_chat_completions))
+        // Anthropic routes
+        .route("/anthropic/v1/messages", post(anthropic_messages))
+        // OpenRouter routes
+        .route("/openrouter/v1/models", get(openrouter_list_models))
+        .route("/openrouter/v1/chat/completions", post(openrouter_chat_completions))
+        // Azure OpenAI routes
+        .route("/azure-openai/chat/completions", post(azure_openai_chat_completions))
+        .route("/azure-openai/completions", post(azure_openai_completions))
+        .route("/azure-openai/embeddings", post(azure_openai_embeddings))
+        .with_state(state)
+}
+
+/// Default HTTP client with a longer timeout suitable for AI requests.
+fn create_client(timeout: Duration) -> reqwest::Client {
+    reqwest::Client::builder()
+        .timeout(timeout)
+        .build()
+        .expect("parameters known to be valid only")
+}
```

Comment:
note: I think it’s better to reuse the client instead of creating a new one for each request. We could use some sort of caching.